### PR TITLE
Fix specs and examples

### PR DESCRIPTION
**Pull request by vkryukov**
Hi, and thanks for creating LangChain - it's a very useful library!

Apologies for unsolicited PR, but I decided to fix some low-hanging spec errors that dialyzer was complaining about, and also fixed `README.md` to use the new return type for `LLMChain.run()`. 

There are still a few dialyzer errors left but they are a bit more cryptic.

Hope that works!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,10 +3,11 @@
 Elixir LangChain enables Elixir applications to integrate AI services and self-hosted models into an application.
 
 Currently supported AI services:
+
 - OpenAI ChatGPT
 - OpenAI DALL-e 2 - image generation
 - Anthropic Claude
-- Google AI - https://generativelanguage.googleapis.com
+- Google AI - <https://generativelanguage.googleapis.com>
```
Comment (brainlid) [inline README.md:10]: Why the surrounding angle brackets?

File: README.md [hunk 2]
```diff
@@ -182,8 +185,8 @@ custom_fn =
   |> LLMChain.run(mode: :while_needs_response)
 
 # print the LLM's answer
-IO.puts(message.content)
-#=> "The hairbrush is located in the drawer."
+IO.puts(update |> ChainResult.to_string())
+# => "The hairbrush is located in the drawer."
```
Comment (brainlid) [inline README.md:189]: Nice!

File: lib/langchain_error.ex [hunk 3]
```diff
@@ -30,7 +30,7 @@ defmodule LangChain.LangChainError do
   Create the exception using either a message or a changeset who's errors are
   converted to a message.
   """
-  @spec exception(message :: String.t() | Ecto.Changeset.t()) :: t() | no_return()
+  @spec exception(message :: String.t() | Ecto.Changeset.t() | keyword()) :: t() | no_return()
```
Comment (brainlid) [inline lib/langchain_error.ex:33]: Thanks!

**Review by brainlid — COMMENTED** [review]
Looks good! Just the one question.

File: README.md (hunk 1 above)
Comment (vkryukov) [reply README.md:10]: Ups - sorry, this is something that `mix format` has inserted, I didn't notice that. (I have a format-on-save setup in my editor). Should we remove this URL alltogether? The other providers don't have a URL, and btw this is not even a valid one (results in a 404).

Comment (vkryukov) [reply README.md]: This commits fixes it: a97885899f32dc06773ef4cd386403e4cbe1d9af (but I'm not sure what's the interaction between the comments and the newly submitted commits)

**Comment by brainlid** [comment]
Thanks!
❤️💛💙💜

**Pull request merged**